### PR TITLE
Update documentation regarding Go+BoringCrypto

### PR DESCRIPTION
**Pull request by jossemargt-3pillar**
Problem: As new contributor is easy to overlook that rke2 componentes
are compiled using a FIPS 140-2 compatible Go compiler.

Solution: Update README file pointing out the Go+BoringCrypto usage.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,15 @@
 # rancher/hardened-build-base
 
+This repository holds the Dockerfiles and builds scripts for [rancher/hardened-build-base](https://hub.docker.com/r/rancher/hardened-build-base) Docker images. The x86_64 image contains a FIPS compatible Go compiler ([Go+BoringCrypto](https://github.com/golang/go/tree/dev.boringcrypto/misc/boring)), used for [compiling rke2 components](https://docs.rke2.io/security/fips_support/#fips-support-in-cluster-components).
+
+Supported architectures are x86_64/amd64, arm64, and s390x.
```
Comment (briandowns) [inline README.md:5]: No arm64 support yet.

File: README.md [hunk 2]
```diff
@@ -1,7 +1,15 @@
 # rancher/hardened-build-base
 
+This repository holds the Dockerfiles and builds scripts for [rancher/hardened-build-base](https://hub.docker.com/r/rancher/hardened-build-base) Docker images. The x86_64 image contains a FIPS compatible Go compiler ([Go+BoringCrypto](https://github.com/golang/go/tree/dev.boringcrypto/misc/boring)), used for [compiling rke2 components](https://docs.rke2.io/security/fips_support/#fips-support-in-cluster-components).
```
Comment (briandowns) [inline README.md:3]: The Go compiler isn't necessarily FIPS 140-2 compliant as much as it contains a FIPS 140-2 compliant crypto module, GoBoring.

Comment (jossemargt-3pillar) [reply README.md:3]: You are right, let me rephrase it.

File: README.md (hunk 1 above)
Comment (briandowns) [reply README.md:5]: That is for trivy and for future-proofing for when we're capable of doing arm64.

File: README.md [hunk 3]
```diff
@@ -1,7 +1,18 @@
...
+- [x86_64/amd64](Dockerfile.amd64)
+- [s390x](Dockerfile.s390x)
+
 ## Build
 
-```sh
-TAG=v1.14.2 make
-```
\ No newline at end of file
+```shell
```
Comment (briandowns) [inline README.md:12]: What effect did this have on the output?

Comment (jossemargt-3pillar) [reply README.md:12]: The output varies depending on the markdown renderer, but in most cases it won't do anything to be honest. I did it mostly to appease my IDE's markdown linter, and I can change it to `sh` and keep it that way for future contributions.

Comment (briandowns) [reply README.md]: Let's keep consistent for now.

Comment (jossemargt-3pillar) [reply README.md]: Sure thing, I just changed it to `sh`

**Review by briandowns — APPROVED** [review]
